src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs

```diff
@@ -46,16 +64,120 @@ int IVsSolutionLoadEvents.OnAfterLoadProjectBatch(bool fIsBackgroundIdleBatch)
 
         int IVsSolutionLoadEvents.OnAfterBackgroundSolutionLoadComplete()
         {
+            // TODO: Not called in DPL scenarios?
+            //if (_inDeferredProjectLoad)
+            //{
+            //    LoadSolutionInBackground();
+            //}
+            //else
+            //{
+                PushCompleteSolutionToWorkspace(s_getProjectInfoForProject);
+            //}
+
+            return VSConstants.S_OK;
+        }
+
+        private async void LoadSolutionInBackground(string solutionFilename)
+        {
+            var componentModel = _serviceProvider.GetService(typeof(SComponentModel)) as IComponentModel;
+            var workspaceProjectContextFactory = componentModel.GetService<IWorkspaceProjectContextFactory>();
+            var solutionInfo = await Task.Run(() => LoadSolutionInfoAsync(solutionFilename), _solutionParsingCancellationTokenSource.Token).ConfigureAwait(true);
+
```

Comment:
threading assert htat we're on the expected thread.  Comment why we use ConfigureAwait(true)